<?xml version="1.0"?>
<launch>
    <arg name="frequency" default="30"/>
    <arg name="optical_flow_topic" default="/mavros/px4flow/raw/optical_flow_rad"/>
    <arg name="imu_topic" default="/mavros/imu/data"/>
    <arg name="odom_topic" default="/foresight/px4flow/odom"/>
    <arg name="odom_frame_id" default="odom"/>
    <arg name="odom_child_frame_id" default="quad/base_link"/>
    <node pkg="foresight" type="px4flow_odom.py" name="px4flow_odom" output="screen">
        <param name="frequency" value="$(arg frequency)"/>
        <param name="optical_flow_topic" value="$(arg optical_flow_topic)"/>
        <param name="imu_topic" value="$(arg imu_topic)"/>
        <param name="odom_topic" value="$(arg odom_topic)"/>
        <param name="odom_frame_id" value="$(arg odom_frame_id)"/>
        <param name="odom_child_frame_id" value="$(arg odom_child_frame_id)"/>
    </node>
</launch>
